Library/ViewModels/PledgeAmountSummaryViewModel.swift

```diff
@@ -0,0 +1,131 @@
+import Foundation
+import KsApi
+import Prelude
+import ReactiveSwift
+
+public protocol PledgeAmountSummaryViewModelInputs {
+  func configureWith(_ project: Project)
+  func viewDidLoad()
+}
+
+public protocol PledgeAmountSummaryViewModelOutputs {
+  var pledgeAmountText: Signal<NSAttributedString, Never> { get }
+  var shippingAmountText: Signal<NSAttributedString, Never> { get }
+  var shippingLocationStackViewIsHidden: Signal<Bool, Never> { get }
+  var shippingLocationText: Signal<String, Never> { get }
+}
+
+public protocol PledgeAmountSummaryViewModelType {
+  var inputs: PledgeAmountSummaryViewModelInputs { get }
+  var outputs: PledgeAmountSummaryViewModelOutputs { get }
+}
+
+public class PledgeAmountSummaryViewModel: PledgeAmountSummaryViewModelType,
+  PledgeAmountSummaryViewModelInputs, PledgeAmountSummaryViewModelOutputs {
+  public init() {
+    let project = Signal.combineLatest(
+      self.projectSignal,
+      self.viewDidLoadProperty.signal
+    )
+    .map(first)
+
+    let backing = project
+      .map { $0.personalization.backing }
+      .skipNil()
+
+    let projectAndBacking = project
+      .zip(with: backing)
+
+    self.pledgeAmountText = projectAndBacking
+      .map { project, backing in
+        attributedCurrency(with: project, amount: backing.pledgeAmount)
+      }
+      .skipNil()
+
+    let shippingAmount = backing
+      .map { Double($0.shippingAmount ?? 0) }
+
+    self.shippingAmountText = project
+      .combineLatest(with: shippingAmount)
+      .map { project, shippingAmount in
+        shippingValue(with: project, with: shippingAmount)
+      }
+      .skipNil()
+
+    self.shippingLocationText = backing
+      .map { $0.locationName }
+      .skipNil()
+      .map { Strings.Shipping_to_country(country: $0) }
+
+    self.shippingLocationStackViewIsHidden = project
+      .map(shouldHideShippingLocationStackView)
+  }
+
+  private let (projectSignal, projectObserver) = Signal<Project, Never>.pipe()
+  public func configureWith(_ project: Project) {
+    self.projectObserver.send(value: project)
+  }
+
+  private let viewDidLoadProperty = MutableProperty(())
+  public func viewDidLoad() {
+    self.viewDidLoadProperty.value = ()
+  }
+
+  public let pledgeAmountText: Signal<NSAttributedString, Never>
+  public let shippingAmountText: Signal<NSAttributedString, Never>
+  public let shippingLocationStackViewIsHidden: Signal<Bool, Never>
+  public let shippingLocationText: Signal<String, Never>
+
+  public var inputs: PledgeAmountSummaryViewModelInputs { return self }
+  public var outputs: PledgeAmountSummaryViewModelOutputs { return self }
+}
+
+private func shouldHideShippingLocationStackView(_ project: Project) -> Bool {
+  guard let backing = project.personalization.backing,
+    let _ = backing.locationName else {
+    return true
+  }
+  if let reward = backing.reward {
```

Comment:
This is interesting. Do you think we need this extra layer of logic? I'd imagine if the `backing.locationName` and `backing.locationId` are not nil that you could assume that the backing was created with a shipping address. 🤔

I wonder if this could cause a bug in a situation where the reward's shipping eligibility is modified by the creator *after* the user has created a pledge (not sure if that's even possible).

Reply:
Another copy-paste, I'll check it out, thanks!

Reply:
Confirmed with team R&F that this cannot occur, so checking that `locationId` is not `nil` on the `Backing` is sufficient. Good catch!